### PR TITLE
feature: payer address

### PR DESCRIPTION
**Pull request by marquinhoalm**
- The address field is a JSON field
- Add hook beforeValidate
- In Validation if all the required fields exist in the request of the `payer_address`
they will be maintained otherwise a default will be use
- Removed address default value to send the provider
- Always will use the `payer_address` saved at database

closes pagarme/superbowleto#236

## Your checklist for this pull request

:rotating_light: Please review this items for a good pull request. :four_leaf_clover:

1. I've read the project's [Contributing Guidelines](../CONTRIBUTING.md)
1. My commits are well written and follow [pagarme/git-style-guide](https://github.com/pagarme/git-style-guide)
1. My changes are well covered by **tests** and **logs**
1. I've updated the project docs (if needed)
1. I fell safe about this implementation
1. I feel comfortable with the code I wrote, and I'm not ashamed to show it to my friends

In a good pull request, everything above is true :relaxed:

### CODE REVIEW

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 0901263843531d3a8f4e542f98db23689a3c85de
  * Your commit message body contains a line that is longer than 72 characters

Check out our Git Style Guide here: https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: ad6d7c1149689be35a2aa01cf868098a43e03f18
  * Your commit message body contains a line that is longer than 72 characters

Check out our Git Style Guide here: https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: b3b34c0224f05b2381440e6c9192befc2011b5d9
  * Your commit message body contains a line that is longer than 72 characters

Check out our Git Style Guide here: https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

File: src/database/migrations/20170807000001-add-payer-address-to-boleto.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { JSON } from 'sequelize'
+
+export default {
+  up: (queryInterface, Sequelize) =>
+    queryInterface.addColumn('Boletos', 'payer_address', {
+      type: JSON
+    }
+  ),
+
+  down: (queryInterface, Sequelize) =>
```
Comment (mccraveiro) [inline src/database/migrations/20170807000001-add-payer-address-to-boleto.js]: Sequelize is defined but never used 
(na linha 4 tb)

File: src/providers/bradesco/index.ts [hunk 2]
```diff
@@ -42,13 +42,13 @@ export const buildPayload = boleto =>
...
-            bairro: always('Vila Olímpia'),
-            cidade: always('São Paulo'),
-            uf: always('SP')
+            cep: path(['payer_address', 'zipcode']),
+            logradouro: path(['payer_address', 'street']),
+            numero: path(['payer_address', 'street_number']),
+            complemento: path(['payer_address', 'complementary']),
+            bairro: path(['payer_address', 'neighborhood']),
+            cidade: path(['payer_address', 'city']),
+            uf: path(['payer_address', 'state'])
```
Comment (mccraveiro) [inline src/providers/bradesco/index.ts]: Linha: 56
```js
.then(applySpec(__, boleto))
```

File: src/resources/boleto/model.ts [hunk 3]
```diff
@@ -1,6 +1,19 @@
...
+  pick,
+  cond,
+  equals,
+  T,
+  identity,
+  always,
+  all,
+  ifElse,
+  has,
+  __
```
Comment (mccraveiro) [inline src/resources/boleto/model.ts]: Tem comma dangle no projeto?

File: src/providers/bradesco/index.ts (hunk 2 above)
Comment (marquinhoalm) [inline src/providers/bradesco/index.ts]: Infelizmente essa solução não deu certo @mccraveiro

File: src/resources/boleto/model.ts (hunk 3 above)
Comment (marquinhoalm) [inline src/resources/boleto/model.ts]: Não temos

File: src/providers/bradesco/index.ts (hunk 2 above)
Comment (mccraveiro) [inline src/providers/bradesco/index.ts]: @marquinhoalm linha 62 agora. Deveria funcionar, não muda nada na lógica. 
@MarcoWorms consegue ver o motivo?

**Pull request merged**